### PR TITLE
Add prelude

### PR DESCRIPTION
**Pull request by tekknolagi**
- Add prelude and boot with it
- Add tests for filter

### CODE REVIEW

File: scrapscript.py [hunk 1]
```diff
@@ -3448,6 +3469,20 @@ def listlength(obj: Object) -> Object:
 }
 
 
+PRELUDE = """
```
Comment (gregorybchris) [inline scrapscript.py:3479]: Perhaps the prelude should be implemented as a dict sort of like stdlib? To allow multiple functions in the prelude?

Comment (gregorybchris) [reply scrapscript.py:3479]: Or a list. Not sure. But not sure how we'd expand the prelude if it's just a string

Comment (tekknolagi) [reply scrapscript.py:3479]: you can use `.` syntax

**Review by gregorybchris — APPROVED** [review]

File: scrapscript.py (hunk 1 above)
Comment (gregorybchris) [reply scrapscript.py:3479]: Ok that's fine. Might prefer a list of strings and then we eval each and add them to the boot env..
But I trust whatever you think is best

File: scrapscript.py [hunk 2]
```diff
@@ -3448,6 +3476,22 @@ def listlength(obj: Object) -> Object:
...
+id = x -> x
+.
+filter = f ->
+| [] -> []
+| [x, ...xs] -> (f x) |> (| #true -> x >+ (filter f xs)
```
Comment (gregorybchris) [inline scrapscript.py]: Parens around (f x) and (| # true ... ) are not needed if you don't want them

Comment (tekknolagi) [reply scrapscript.py]: alright let's go wild

**Pull request merged**